tests/tom-swifty-test/SwiftReflector/ExpermimentalTests.cs

```diff
@@ -0,0 +1,255 @@
+@frozen
+public struct OneInt {
+	public var X:Int
+	public init () { X = 17 }
+}
+";
+
+			// C# aux code:
+			// public struct XXOpaque1 {
+			// nint x0
+			// public static unsafe nint Getter (OneInt self)
+			// {
+			//    fixed (byte* thisSwiftDataPtr = SwiftRuntimeLibrary.SwiftMarshal.StructMarshal.Marshaler.PrepareValueType (self)) {
+			//        var p = (XXOpaque1 *)thisSwiftDataPtr;
+			//        return XGetter (*p);
+			// }
+			// [DllImport ("libExpermimentalTests.dylib", EntryPoint = "_$s18ExpermimentalTests6OneIntV1XSivg")]
+			// public static extern nint XGetter (XXOpaque1 r);
+			//
+
+			var xxopName = new CSIdentifier ("XXOpaque1");
+			var selfID = (CSIdentifier)"self";
+			var thisSwiftDataPtrID = (CSIdentifier)"thisSwiftDataPtr";
+			var pID = (CSIdentifier)"p";
+			var getterID = (CSIdentifier)"Getter";
+			var xgetterID = (CSIdentifier)"XGetter";
+
+			var st = new CSStruct (CSVisibility.Public, xxopName);
+			st.Fields.Add (CSFieldDeclaration.FieldLine (CSSimpleType.NInt, "x0", value: null, CSVisibility.None));
+			var body = new CSCodeBlock ();
+			var fixedBlock = new CSFixedCodeBlock (CSSimpleType.ByteStar, thisSwiftDataPtrID,
+				new CSFunctionCall ("SwiftRuntimeLibrary.SwiftMarshal.StructMarshal.Marshaler.PrepareValueType", false, selfID), null);
+
+			fixedBlock.Add (CSVariableDeclaration.VarLine (pID, new CSCastExpression (new CSSimpleType ($"{xxopName.Name}*"), thisSwiftDataPtrID)));
+			fixedBlock.Add (CSReturn.ReturnLine (new CSFunctionCall (xgetterID.Name, false, new CSUnaryExpression (CSUnaryOperator.Indirection, pID))));
+			body.Add (fixedBlock);
+
+			var getter = new CSMethod (CSVisibility.Public, CSMethodKind.StaticUnsafe, CSSimpleType.NInt, getterID,
+				new CSParameterList (new CSParameter (new CSSimpleType ("OneInt"), selfID)), body);
+			st.Methods.Add (getter);
+
+
+			var pinvoke = CSMethod.PInvoke (CSVisibility.Public, CSSimpleType.NInt, xgetterID.Name,
+				CSConstant.Val ("libExpermimentalTests.dylib"), "$s18ExpermimentalTests6OneIntV1XSivg", new CSParameterList (new CSParameter (new CSSimpleType (xxopName.Name), (CSIdentifier)"r")));
+			st.Methods.Add (pinvoke);
+
+			var decl = CSVariableDeclaration.VarLine ("oneInt", CSFunctionCall.Ctor ("OneInt"));
+			var printerGenerated = CSFunctionCall.ConsoleWriteLine ((CSIdentifier)"oneInt.X");
+			var printerDerived = CSFunctionCall.ConsoleWriteLine (new CSFunctionCall ($"{xxopName.Name}.{getterID.Name}", false, (CSIdentifier)"oneInt"));
+			var callingCode = CSCodeBlock.Create (decl, printerGenerated, printerDerived);
+			TestRunning.TestAndExecute (swiftCode, callingCode, "17\n17\n", otherClass: st);
+		}
+
+		[Test]
+		public void ByteSingleIntStruct ()
+		{
+			var swiftCode = @"
+@frozen
+public struct OneInt {
+	public var A:Int8
+	public var X:Int
+	public init () { A = 7; X = 17 }
+}
+";
+
+			// C# aux code:
+			// public struct XXOpaque2 {
+			// nint x0
+			// nint x1
+			// public static unsafe nint Getter (OneInt self)
```

Comment:
What do x0 and x1 represent for the remainder of the test cases

Reply:
In the current implementation of BTfS, the payload for value types is an array and we always pass it by reference. The reason is that the Swift ABI allows for value types to be passed by value that are up to 4 machine words (consuming 4 registers), but the standard ABI only allows for 2. This requires me to write adapters for all methods and functions that take value types. We don't like this because it's really inefficient. The runtime team is working on this so I can improve the marshaling. So, I made opaque structs that just take up space which I can use in pinvokes by blitting the payload array contents directly into them. I defined these two just for the sake of the test to answer the question "will this even work". Since my runtime currently stops at 2 machine words for pass by value, this is where I did the cutoff.
Make sense?